lesson4/argirovatina_hometask.rb

```diff
@@ -11,7 +11,8 @@ def lazy?(key)
   end
 
   def respond_to_missing?(method_name, include_private = false)
-    self.has_key? method_name || super
+    key = get_correct_key(method_name)
```

Comment:
I guess I understood. You are trying to eliminate case about string/symbol differences. But I told you, that `#get_correct_key` in that case is overkill because it raises an unnecessary exception. You can create pair of methods: `#correct_key` and `#correct_key!`. First one would check just presence of a key in hash (but not to raise exception), second one would use first one and raise exception if first one returns `nil`.